### PR TITLE
[Communication][Email] Preparing SDK for GA Release

### PR DESCRIPTION
**Pull request by yogeshmo**
- The SDK was regenerated with the new GA-ready swagger
- Versioning was updated
- Test recordings were updated
- The CHANGELOG was updated

### CODE REVIEW

File: sdk/communication/azure-communication-email/CHANGELOG.md [hunk 1]
```diff
@@ -1,14 +1,10 @@
...
+## 1.0.0 (2023-03-31)
 
-### Features Added
-
-### Breaking Changes
-
-### Bugs Fixed
-
-### Other Changes
+The public release of the Azure Communication Services SDK for Email has the following features:
```
Comment (srnagar) [inline sdk/communication/azure-communication-email/CHANGELOG.md:5]: This should follow the format that is shown on the left hand side with the sub-sections.

File: sdk/communication/azure-communication-email/README.md [hunk 2]
```diff
@@ -54,7 +54,7 @@ add the direct dependency to your project as follows.
 <dependency>
     <groupId>com.azure</groupId>
     <artifactId>azure-communication-email</artifactId>
-    <version>1.0.0-beta.1</version>
+    <version>1.0.0/version>
 </dependency>
 ```
 ## Key concepts
```
Comment (srnagar) [inline sdk/communication/azure-communication-email/README.md:60]:
```suggestion
[//]: # ({x-version-update-end})

## Key concepts
```

File: sdk/communication/azure-communication-email/pom.xml [hunk 3]
```diff
@@ -12,7 +12,7 @@
 
   <groupId>com.azure</groupId>
   <artifactId>azure-communication-email</artifactId>
-  <version>1.0.0-beta.3</version> <!-- {x-version-update;com.azure:azure-communication-email;current} -->
+  <version>1.0.0</version> <!-- {x-version-update;com.azure:azure-communication-email;current} -->
```
Comment (srnagar) [inline sdk/communication/azure-communication-email/pom.xml:15]: The version should be updated in `eng/versioning/version_client.txt` too.

File: sdk/communication/azure-communication-email/CHANGELOG.md (hunk 1 above)
Comment (yogeshmo) [reply sdk/communication/azure-communication-email/CHANGELOG.md:5]: There aren't any changes between this version and the previous preview version. Should I just leave these sections in here, but blank? I vaguely remember getting an error in the release pipeline when I left a section empty last time.

File: sdk/communication/azure-communication-email/pom.xml (hunk 3 above)
Comment (srnagar) [reply sdk/communication/azure-communication-email/pom.xml:15]: This is not resolved yet. I don't see the updates to `eng/versioning/version_client.txt`.

**Review by srnagar — APPROVED** [review]
LGTM

**Pull request merged**